components/export/api/src/main/java/org/phenotips/export/internal/DataToCellConverter.java

```diff
@@ -283,100 +287,222 @@ public DataSection genesHeader() throws Exception
         int y = 0;
 
-        if (present.contains("candidate")) {
-            PatientData<Map<String, String>> candidateGenes = patient.getData("genes");
-            if (candidateGenes != null && candidateGenes.isIndexed()) {
-                DataCell cell = new DataCell("Candidate", 0, y);
+        PatientData<Map<String, String>> variants = patient.getData("variants");
+        // empties should be created in the case that there are no variants to write
+        if (variants == null || !variants.isIndexed()) {
+            /* Genesymbol and variant cDNA columns are always present */
+            Integer emptyX = 0;
+            for (int i = 0; i < present.size(); i++) {
+                DataCell cell = new DataCell("", emptyX, y);
                 section.addCell(cell);
-                for (Map<String, String> candidateGene : candidateGenes) {
-                    hasGenes = true;
-                    String geneName = candidateGene.get("gene");
-                    cell = new DataCell(geneName, 1, y);
-                    section.addCell(cell);
+                emptyX++;
+            }
+            return section;
+        }
+
+        List<String> fields =
+            Arrays.asList("protein", "transcript", "dbsnp", "zygosity", "effect",
+                "interpretation", "inheritance",
+                "evidence", "segregation", "sanger", "resolution");
+        List<String> translatables =
+            Arrays.asList("effect", "interpretation", "inheritance", "segregation", "sanger", "resolution");
+
+        Map<String, String> valueTranslates = new HashMap<String, String>();
+        valueTranslates.put("likely_pathogenic", "Likely Pathogenic");
+        valueTranslates.put("likely_benign", "Likely Benign");
+        valueTranslates.put("variant_u_s", "Variant of Unknown Significance");
+        valueTranslates.put("in_progress", "In progress");
+        valueTranslates.put("not_done", "Not done");
+        valueTranslates.put("solved", "Confirmed causal");
+        valueTranslates.put("denovo_germline", "de novo germline");
+        valueTranslates.put("denovo_s_mosaicism", "de novo somatic mosaicism");
+        valueTranslates.put("insertion_in_frame", "insertion - in frame");
+        valueTranslates.put("deletion_in_frame", "deletion - in frame");
+        valueTranslates.put("deletion_frameshift", "deletion - frameshift");
+        valueTranslates.put("insertion_frameshift", "insertion - frameshift");
+        valueTranslates.put("indel_in_frame", "indel - in frame");
+        valueTranslates.put("indel_frameshift", "indel - frameshift");
+        valueTranslates.put("repeat_expansion", "repeat expansion");
+        List<String> upperCase =
+            Arrays.asList("pathogenic", "benign", "complete", "rejected", "unknown", "missense", "nonsense",
+                "duplication", "synonymous", "other", "maternal", "paternal");
+
+        for (Map<String, String> variant : variants) {
+            int j = 0;
```

Comment:
you use `i` and `j` and `hX` and `emptyX` as a counter, interchangeably. You should pick one and stick with it.